OpenChange/MAPIStoreAppointmentWrapper.m

```diff
@@ -1725,11 +1726,16 @@ - (void) _computeGlobalObjectIds
   NSData *binPrefix;
   TALLOC_CTX *localMemCtx;
 
-  localMemCtx = talloc_zero (NULL, TALLOC_CTX);
+  localMemCtx = talloc_zero(NULL, TALLOC_CTX);
 
   memset (&newGlobalId, 0, sizeof (struct GlobalObjectId));
 
   uid = [event uid];
```

Comment:
nil is `(id)0`
null is `(void *)0`

so `nil == NULL` is `(id)0 == (id)(void*)0`

A proper compiler probably will complain about `uid == NULL` mismatch types comparison. 

TL;DR: `if (!uid)`